api/src/pcapi/core/offerers/commands.py

```diff
@@ -135,19 +136,29 @@ def synchronize_accessibility_with_acceslibre(dry_run: bool = False, force_sync:
         for venue in venues_list:
             slug = venue.accessibilityProvider.externalAccessibilityId
             last_update = accessibility_provider.get_last_update_at_provider(slug=slug)
-            if last_update or not venue.accessibilityProvider.externalAccessibilityData:
-                if force_sync or venue.accessibilityProvider.lastUpdateAtProvider < last_update:
-                    venue.accessibilityProvider.lastUpdateAtProvider = last_update
-                    accessibility_data = accessibility_provider.get_accessibility_infos(
-                        slug=venue.accessibilityProvider.externalAccessibilityId
-                    )
-                    venue.accessibilityProvider.externalAccessibilityData = (
-                        accessibility_data.dict() if accessibility_data else None
-                    )
-                    db.session.add(venue.accessibilityProvider)
+            # if last_update is not None: match still exist
+            # Then we update accessibility data if :
+            # 1. accessibility data is None
+            # 2. we have forced the synchronization
+            # 3. accessibility data has been updated on acceslibre side
+            if last_update and (
+                not venue.accessibilityProvider.externalAccessibilityData
+                or force_sync
+                or venue.accessibilityProvider.lastUpdateAtProvider.astimezone(pytz.utc)
+                < last_update.astimezone(pytz.utc)
+            ):
+                venue.accessibilityProvider.lastUpdateAtProvider = last_update
+                accessibility_data = accessibility_provider.get_accessibility_infos(
+                    slug=venue.accessibilityProvider.externalAccessibilityId
+                )
+                venue.accessibilityProvider.externalAccessibilityData = (
+                    accessibility_data.dict() if accessibility_data else None
+                )
+                db.session.add(venue.accessibilityProvider)
 
-            # If slug has not been found at acceslibre, we try to find a new match
-            else:
+            # if last_update is None, the slug has been removed from acceslibre, we try a new match
+            # and save accessibility data to DB
+            if not last_update:
```

Comment:
Peut-être elif au lieu de if ?

Reply:
le premier if check plusieurs choses, pas seulement `last_update`. Si on a un `last_update` mais pas 
- accessibility data is None
 - force_sync
- une last_update plus récente

-> on va quand même rentrer dans le `elif`, alors qu'il est là que pour récupérer un nouveau match si on a perdu l'ancien